### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by CurryNRice**
Rewrote most functionallity to better fit to UML design

### CODE REVIEW

File: core/src/main/java/com/g16/feyrune/model/creature/BaseCreature.java [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
+import java.util.ArrayList;
+import java.util.List;
 
 import static java.lang.Math.min;
 
-public class BaseCreature {
+public class BaseCreature implements ICreature {
     //Stats variables
     private final double maxHealth;
     private double health;
```
Comment (CurryNRice) [inline core/src/main/java/com/g16/feyrune/model/creature/BaseCreature.java:16]: health shold be int!!!!!!!!!!!!😡

File: core/src/main/java/com/g16/feyrune/model/combat/actions/attackAction.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+public class attackAction implements ICombatAction {
+
+    @Override
+    public boolean executeMove(ICombatCreature actor, ICombatCreature target) {
+        int moveSelection = Random.random.nextInt(actor.getMoves().size());
+        AttackHandler.handleAttack(actor,target,actor.getMoves().get(moveSelection));
+        return false;
+    }
+}
```
Comment (Krysset) [inline core/src/main/java/com/g16/feyrune/model/combat/actions/attackAction.java:16]: Should not be in actions package

File: core/src/main/java/com/g16/feyrune/Util/Random.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package com.g16.feyrune.Util;
+
+public class Random {
+    public static final java.util.Random random=new java.util.Random();
+}
```
Comment (Krysset) [inline core/src/main/java/com/g16/feyrune/Util/Random.java:5]: Should be using a public interface to interact with random instead

File: core/src/main/java/com/g16/feyrune/model/combat/creatures/EnemyCreature.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+public class EnemyCreature implements ICombatCreature {
+    private ICreature creature;
+
+    public EnemyCreature(ICreature creature) {
+        this.creature = creature;
+    }
+
+    @Override
+    public ICombatAction selectAction(ICombatCreature actor, ICombatCreature target) {
+        return null;
```
Comment (x183) [inline core/src/main/java/com/g16/feyrune/model/combat/creatures/EnemyCreature.java:20]: No null!

File: core/src/main/java/com/g16/feyrune/model/creature/CreatureFactory.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import com.g16.feyrune.interfaces.ICreature;
+import com.g16.feyrune.model.action.BaseAttack;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+import com.g16.feyrune.Util.Pair;
+
+public class CreatureFactory {
+    /**
+     *
+     * @return a new Object of type ICreature
+     */
```
Comment (x183) [inline core/src/main/java/com/g16/feyrune/model/creature/CreatureFactory.java:12]: Whoops! this code shouldn't be here

File: core/src/main/java/com/g16/feyrune/model/overworld/MovementHandler.java [hunk 6]
```diff
@@ -1,25 +1,16 @@
...
-import com.g16.feyrune.model.map.parser.Map;
+import com.g16.feyrune.model.overworld.map.Map;
 
 import java.awt.*;
 
 public class MovementHandler {
-    private Player player;
     private long moveFrequency = 500;
     private long lastMoved = 0;
     private int dirX = 0;
```
Comment (x183) [inline core/src/main/java/com/g16/feyrune/model/overworld/MovementHandler.java:11]: Why isn't this a Point? Point has built in functionality for moving

File: core/src/main/java/com/g16/feyrune/model/overworld/OverworldModel.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        this.map = Map.getGlobalMap();
+    }
+
+    public void update(){
+        movePlayer();
+    }
+
+    public void movePlayer() {
+        Point deltaPos = movementHandler.calculateMovement(player.getCoordinates(), map);
+        player.move(deltaPos.x, deltaPos.y);
```
Comment (x183) [inline core/src/main/java/com/g16/feyrune/model/overworld/OverworldModel.java:25]: Why not just give it a copy of point?

File: core/src/main/java/com/g16/feyrune/model/overworld/encounter/EncounterHandler.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+public class EncounterHandler {
+    public EncounterHandler() {
+    }
+
+    /**
+     *
+     * @return
+     */
+    public Encounter createEncounter(Pair<Integer, Integer>[] monsterList) {
```
Comment (x183) [inline core/src/main/java/com/g16/feyrune/model/overworld/encounter/EncounterHandler.java:17]: Whoops, this code shouldn't be here

File: core/src/main/java/com/g16/feyrune/model/player/Player.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Player {
+    private String name;
+    private Point coordinates;
+    private Texture texture;
+    private ICreature creature; //TODO: this is wrong and temp
```
Comment (x183) [inline core/src/main/java/com/g16/feyrune/model/player/Player.java:15]: Yes, wrong?

File: core/src/main/java/com/g16/feyrune/view/player/PlayerRenderer.java [hunk 10]
```diff
@@ -39,7 +39,7 @@ public PlayerRenderer(Player player) {
         idleLeftAnimation = AnimationUtils.getAnimationFrames(texture, 8, 6, 4, 4);
         currentAnimation = walkRightAnimation;
         this.player = player;
-        playerPos = new Vector2(player.getPosX(), player.getPosY());
+        playerPos = new Vector2(player.getCoordinates().x, player.getCoordinates().y);
```
Comment (x183) [inline core/src/main/java/com/g16/feyrune/view/player/PlayerRenderer.java:42]: Should be a point and not a Vector2

**Review by x183 — COMMENTED** [review]
Some stuff left to fix before merging

File: core/src/main/java/com/g16/feyrune/Util/Random.java (hunk 3 above)
Comment (x183) [reply core/src/main/java/com/g16/feyrune/Util/Random.java]: Is somewhat fixed now

File: core/src/main/java/com/g16/feyrune/model/TimeService.java [hunk 11]
```diff
@@ -5,13 +5,12 @@
...
 
     public static void initialize(){
         startTime = TimeUtils.millis();
     }
 
     public static long getElapsedTime(){
-        return elapsedTime = TimeUtils.timeSinceMillis(startTime);
+        return TimeUtils.timeSinceMillis(startTime);
     }
 }
```
[the remark this answers is not in the document]
Comment (CurryNRice) [reply core/src/main/java/com/g16/feyrune/model/TimeService.java:16]: no

**Review by x183 — APPROVED** [review]
Seems good

**Review by Supergamer1337 — APPROVED** [review]
I approve

**Pull request merged**